### PR TITLE
3.0: Handle differences between AWS Batch and Slurm in API tests

### PR DESCRIPTION
**Pull request by ermanno**
### Notes
When Slurm clusters are created, the instances are ready as soon as the stack creation is completed. With AWS Batch this is not true, so we add a step to wait until the desired number of instances are present in the cluster. Furthermore, the queue name for AWS Batch clusters is not populated (not that we can specify only one queue), and so this patch handles also that corner case when creating the compute_node_map.

### Tests
Run API integration tests.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/integration-tests/tests/api/test_api.py [hunk 1]
```diff
@@ -99,10 +101,18 @@ def _cloudformation_wait(region, stack_name, status):
...
+
 def _ec2_wait_terminated(region, instances):
+    _ec2_wait(region, instances, "terminated")
+
+
+def _ec2_wait(region, instances, status):
     config = Config(region_name=region)
     ec2 = boto3.client("ec2", config=config)
-    waiter = ec2.get_waiter("instance_terminated")
+    waiter = ec2.get_waiter(f"instance_{status}")
```
Comment (lukeseawalker) [inline tests/integration-tests/tests/api/test_api.py:115]: minor: since this is a generic `get_waiter` and not just a getter for `instance_*` waiters, I would keep this generic, and pass the full waiter name from the outside

File: tests/integration-tests/tests/api/test_api.py [hunk 2]
```diff
@@ -220,6 +237,10 @@ def _test_stop_compute_fleet(region, cluster_compute_fleet_client, cluster_insta
     compute_node_map = _test_describe_cluster_compute_nodes(region, cluster_instances_client, cluster_name)
     instances_to_terminate = _get_instances_to_terminate(compute_node_map)
 
+    # EC2 instances in "running" or "stopping" state cannot be terminated, so we will wait for the instances we
```
Comment (lukeseawalker) [inline tests/integration-tests/tests/api/test_api.py:240]: comment is confusing, you are stating that instance in "running" cannot be terminated and then you wait instance to be running to terminate them. Can you please clarify?

File: tests/integration-tests/tests/api/test_api.py [hunk 3]
```diff
@@ -187,15 +201,18 @@ def _test_describe_cluster_compute_nodes(region, client, cluster_name, all_termi
 
 def _add_compute_nodes(instances, compute_node_map):
     for instance in instances:
-        if instance.queue_name not in compute_node_map:
-            compute_node_map[instance.queue_name] = set()
-        compute_node_map[instance.queue_name].add(instance.instance_id)
+        # The AWS Batch queue name is not populated
+        queue_name = instance.get("queue_name", "aws_queue")
```
Comment (lukeseawalker) [inline tests/integration-tests/tests/api/test_api.py:205]: what is "aws_queue"? did you mean "awsbatch_queue"?

File: tests/integration-tests/tests/api/test_api.py (hunk 1 above)
Comment (ermanno) [reply tests/integration-tests/tests/api/test_api.py:115]: Agreed.

File: tests/integration-tests/tests/api/test_api.py (hunk 2 above)
Comment (ermanno) [reply tests/integration-tests/tests/api/test_api.py:240]: Typo, I meant to write "pending".

File: tests/integration-tests/tests/api/test_api/test_cluster_slurm/pcluster.config.update.yaml [hunk 4]
```diff
@@ -1,75 +1,30 @@
 Image:
   Os: {{ os }}
-Tags:
```
Comment (hanwen-pcluste) [inline tests/integration-tests/tests/api/test_api/test_cluster_slurm/pcluster.config.update.yaml:3]: Why would the test be simplified?

**Review by hanwen-pcluste — COMMENTED** [review]
Great work!

File: tests/integration-tests/tests/api/test_api/test_cluster_slurm/pcluster.config.update.yaml (hunk 4 above)
Comment (ermanno) [reply tests/integration-tests/tests/api/test_api/test_cluster_slurm/pcluster.config.update.yaml:3]: The Slurm test was unnecessarily complex for what we intended to test (I originally copied it from the configuration used for the update config tests), I reduced it to a more "basic" case that is sufficient for our purposes.

**Review by demartinofra — APPROVED** [review]
